activerecord/test/activejob/destroy_async_job_not_present_test.rb

```diff
@@ -10,22 +10,36 @@ class UnusedBelongsTo < ActiveRecord::Base
   self.destroy_association_async_job = nil
 end
 
+class UndefinedConstantBelongsTo < ActiveRecord::Base
+  self.destroy_association_async_job = "UndefinedConstantJob"
+end
+
 class ActiveJobNotPresentTest < ActiveRecord::TestCase
   test "has_one dependent destroy_async requires activejob" do
-    assert_raises ActiveRecord::ActiveJobRequiredError do
+    error = assert_raises ActiveRecord::ConfigurationError do
       UnusedDestroyAsync.has_one :unused_belongs_to, dependent: :destroy_async
     end
+    assert_equal "A valid destroy_association_async_job is required to use `dependent: :destroy_async` on associations", error.message
```

Comment:
This is just my personal preference, but when checking error messages I try not to overspecify.  That way error messages can be updated / improved without having to update tests that are semantically the same.  So, for example:

```suggestion
    assert_match %r/destroy_association_async_job/, error.message
```

Might be sufficient to assert that the `ActiveRecord::ConfigurationError` is indeed about a missing `destroy_association_async_job` value.  `%r/dependent: :destroy_async/` might also work, too.